diff --git a/plugins/Ubuntu/Settings/Fingerprint/CMakeLists.txt b/plugins/Ubuntu/Settings/Fingerprint/CMakeLists.txt
index b0ff9e9..d653a86 100644
--- a/plugins/Ubuntu/Settings/Fingerprint/CMakeLists.txt
+++ b/plugins/Ubuntu/Settings/Fingerprint/CMakeLists.txt
@@ -11,5 +11,5 @@ add_library(UbuntuSettingsFingerprintQml MODULE
     ubuntusettingsfingerprint.cpp
     ${assets}
 )
-qt5_use_modules(UbuntuSettingsFingerprintQml Core Qml Quick)
+target_link_libraries(UbuntuSettingsFingerprintQml Qt5::Core Qt5::Qml Qt5::Quick)
 add_usc_plugin(Ubuntu.Settings.Fingerprint 0.1 Ubuntu/Settings/Fingerprint TARGETS UbuntuSettingsFingerprintQml)
diff --git a/plugins/Ubuntu/Settings/Vpn/CMakeLists.txt b/plugins/Ubuntu/Settings/Vpn/CMakeLists.txt
index da829c9..442ede7 100644
--- a/plugins/Ubuntu/Settings/Vpn/CMakeLists.txt
+++ b/plugins/Ubuntu/Settings/Vpn/CMakeLists.txt
@@ -18,6 +18,6 @@ add_library(UbuntuSettingsVpn MODULE
     ubuntusettingsvpn.cpp
 )
 
-qt5_use_modules(UbuntuSettingsVpn Core Qml Quick)
+target_link_libraries(UbuntuSettingsVpn Qt5::Core Qt5::Qml Qt5::Quick)
 
 add_usc_plugin(Ubuntu.Settings.Vpn 0.1 Ubuntu/Settings/Vpn TARGETS UbuntuSettingsVpn)
