diff --git a/docs/qml/CMakeLists.txt b/docs/qml/CMakeLists.txt
index 19976ab2..79bcf054 100644
--- a/docs/qml/CMakeLists.txt
+++ b/docs/qml/CMakeLists.txt
@@ -3,7 +3,7 @@ FIND_PROGRAM(QDOC_EXECUTABLE qdoc)
 if(QDOC_EXECUTABLE)
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/ubuntu-download-manager.qdocconf.in ${CMAKE_CURRENT_BINARY_DIR}/ubuntu-download-manager.qdocconf @ONLY)
 add_custom_target(qmldoc
-${QDOC_EXECUTABLE} -qt=qt5 ${CMAKE_CURRENT_BINARY_DIR}/ubuntu-download-manager.qdocconf
+${QDOC_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/ubuntu-download-manager.qdocconf
 WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
 COMMENT "Generating QML API documentation with qdoc" VERBATIM
 )
