diff --git a/src/downloads/test-daemon/main.cpp b/src/downloads/test-daemon/main.cpp
index 8d34311c..1504378c 100644
--- a/src/downloads/test-daemon/main.cpp
+++ b/src/downloads/test-daemon/main.cpp
@@ -26,7 +26,7 @@
 #define DAEMON_PATH "-daemon-path"
 
 int main(int argc, char *argv[]) {
-    QStandardPaths::enableTestMode(true);
+    QStandardPaths::setTestModeEnabled(true);
     QCoreApplication a(argc, argv);
 
     // similar to a real daemon but allows to force certain cases, for example,
diff --git a/tests/common/base_testcase.cpp b/tests/common/base_testcase.cpp
index 204845b4..2fcfe2d0 100644
--- a/tests/common/base_testcase.cpp
+++ b/tests/common/base_testcase.cpp
@@ -92,7 +92,7 @@ BaseTestCase::removeDir(const QString& dirName) {
 void
 BaseTestCase::init() {
 //    qInstallMessageHandler(noMessageOutput);
-    QStandardPaths::enableTestMode(true);
+    QStandardPaths::setTestModeEnabled(true);
 }
 
 void
