diff --git a/libqmenumodel/QMenuModel/CMakeLists.txt b/libqmenumodel/QMenuModel/CMakeLists.txt
index eb4d463..7f0f74d 100644
--- a/libqmenumodel/QMenuModel/CMakeLists.txt
+++ b/libqmenumodel/QMenuModel/CMakeLists.txt
@@ -13,14 +13,15 @@ include_directories(
     ${GLIB_INCLUDE_DIRS}
 )
 
+find_package(Qt5Qml REQUIRED)
+find_package(Qt5Quick REQUIRED)
 target_link_libraries(qmenumodel-qml
     qmenumodel
     ${GLIB_LDFLAGS}
     ${GIO_LDFLAGS}
+    Qt5::Qml Qt5::Quick
 )
 
-qt5_use_modules(qmenumodel-qml Qml Quick)
-
 execute_process(COMMAND ${CMAKE_COMMAND} -E copy_if_different "${CMAKE_CURRENT_SOURCE_DIR}/qmldir"
                                                               "${CMAKE_CURRENT_BINARY_DIR}/qmldir")
 
diff --git a/libqmenumodel/src/CMakeLists.txt b/libqmenumodel/src/CMakeLists.txt
index e8eb4c0..9a49840 100644
--- a/libqmenumodel/src/CMakeLists.txt
+++ b/libqmenumodel/src/CMakeLists.txt
@@ -45,13 +45,13 @@ include_directories(
     ${GIO_INCLUDE_DIRS}
 )
 
+find_package(Qt5Quick REQUIRED)
 target_link_libraries(${SHAREDLIBNAME}
     ${GLIB_LDFLAGS}
     ${GIO_LDFLAGS}
+    Qt5::Core Qt5::Qml Qt5::Quick
 )
 
-qt5_use_modules(${SHAREDLIBNAME} Core Qml Quick)
-
 install(TARGETS ${SHAREDLIBNAME} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 set(QMENUMODEL_HEADERS
