From b3718180350156b2858cd599c97df381549924fa Mon Sep 17 00:00:00 2001
From: Ivan Semkin <ivan@semkin.ru>
Date: Mon, 17 Sep 2018 14:54:13 +0300
Subject: [PATCH] Explicitly require c++14 support

Fixes #11
---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index af0c7b19..8e69cc77 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,6 @@
 cmake_minimum_required(VERSION 2.8.10)
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 # Default install location. Must be set here, before setting the project.
 if (NOT DEFINED CMAKE_INSTALL_PREFIX)
