diff --git a/include/unity/scopes/OnlineAccountClient.h b/include/unity/scopes/OnlineAccountClient.h
index e1f9d2b9..90ef4f82 100644
--- a/include/unity/scopes/OnlineAccountClient.h
+++ b/include/unity/scopes/OnlineAccountClient.h
@@ -23,6 +23,7 @@
 #include <unity/scopes/Result.h>
 #include <unity/util/NonCopyable.h>
 
+#include <functional>
 #include <memory>
 
 namespace unity
diff --git a/include/unity/scopes/internal/AbstractObject.h b/include/unity/scopes/internal/AbstractObject.h
index d66e8018..2c30d48c 100644
--- a/include/unity/scopes/internal/AbstractObject.h
+++ b/include/unity/scopes/internal/AbstractObject.h
@@ -21,6 +21,7 @@
 #include <unity/util/DefinesPtrs.h>
 #include <unity/util/NonCopyable.h>
 
+#include <functional>
 #include <mutex>
 
 namespace unity
diff --git a/include/unity/scopes/internal/Reaper.h b/include/unity/scopes/internal/Reaper.h
index df2d3707..53b4846a 100644
--- a/include/unity/scopes/internal/Reaper.h
+++ b/include/unity/scopes/internal/Reaper.h
@@ -23,6 +23,7 @@
 
 #include <atomic>
 #include <condition_variable>
+#include <functional>
 #include <list>
 #include <mutex>
 #include <thread>
diff --git a/include/unity/scopes/internal/ReplyImpl.h b/include/unity/scopes/internal/ReplyImpl.h
index d6c0bc15..a8769d2f 100644
--- a/include/unity/scopes/internal/ReplyImpl.h
+++ b/include/unity/scopes/internal/ReplyImpl.h
@@ -26,6 +26,7 @@
 #include <unity/scopes/Variant.h>
 
 #include <atomic>
+#include <functional>
 
 namespace unity
 {
diff --git a/include/unity/scopes/internal/zmq_middleware/ServantBase.h b/include/unity/scopes/internal/zmq_middleware/ServantBase.h
index cc47db8d..7bcc8b2d 100644
--- a/include/unity/scopes/internal/zmq_middleware/ServantBase.h
+++ b/include/unity/scopes/internal/zmq_middleware/ServantBase.h
@@ -23,6 +23,7 @@
 #include <unity/scopes/internal/zmq_middleware/Current.h>
 #include <unity/util/NonCopyable.h>
 
+#include <functional>
 #include <memory>
 #include <string>
 #include <unordered_map>
diff --git a/include/unity/scopes/qt/internal/QScopeBaseAPIImpl.h b/include/unity/scopes/qt/internal/QScopeBaseAPIImpl.h
index 7657a424..81ff2aa7 100644
--- a/include/unity/scopes/qt/internal/QScopeBaseAPIImpl.h
+++ b/include/unity/scopes/qt/internal/QScopeBaseAPIImpl.h
@@ -28,6 +28,7 @@
 
 #include <QtCore/QObject>
 
+#include <functional>
 #include <thread>
 #include <memory>
 #include <atomic>
