--- gtest/cmake_install.cmake.orig	2018-06-19 23:51:50.000000000 +0300
+++ gtest/cmake_install.cmake	2018-06-20 00:07:48.884721083 +0300
@@ -46,6 +46,6 @@
 endif()
 
 if("x${CMAKE_INSTALL_COMPONENT}x" STREQUAL "xUnspecifiedx" OR NOT CMAKE_INSTALL_COMPONENT)
-  file(INSTALL DESTINATION "${CMAKE_INSTALL_PREFIX}/include" TYPE DIRECTORY FILES "/usr/src/gtest/include/gtest")
+  file(INSTALL DESTINATION "${CMAKE_INSTALL_PREFIX}/include" TYPE DIRECTORY FILES "/usr/src/gtest")
 endif()
 
