diff --git a/var/lib/lxc/android/config b/var/lib/lxc/android/config
index 912b72c..af33a23 100644
--- a/var/lib/lxc/android/config
+++ b/var/lib/lxc/android/config
@@ -1,18 +1,17 @@
-lxc.rootfs = /var/lib/lxc/android/rootfs
-lxc.utsname = armhf
+lxc.rootfs.path = /var/lib/lxc/android/rootfs
+lxc.uts.name = armhf
 
-lxc.network.type = none
+lxc.net.0.type = none
 
-lxc.devttydir = lxc
-lxc.tty = 4
-lxc.pts = 1024
+lxc.tty.dir = lxc
+lxc.tty.max  = 4
+lxc.pty.max = 1024
 lxc.arch = armhf
 lxc.cap.drop = mac_admin mac_override
-lxc.pivotdir = lxc_putold
 
 lxc.hook.pre-start = /var/lib/lxc/android/pre-start.sh
 
-lxc.init_cmd = /init
+lxc.init.cmd = /init
 
-lxc.aa_profile = unconfined
+lxc.apparmor.profile = unconfined
 lxc.autodev = 0
