diff --git a/CMakeLists.txt b/CMakeLists.txt
index a2270f60..41da2a66 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,7 +28,14 @@ endif()
 
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x -fno-permissive -pedantic -Wall -Wextra")
+find_package(Qt5Concurrent REQUIRED)
+find_package(Qt5DBus REQUIRED)
+find_package(Qt5Qml REQUIRED)
+find_package(Qt5Quick REQUIRED)
+find_package(Qt5Sql REQUIRED)
+find_package(Qt5Test REQUIRED)
 find_package(Qt5Widgets REQUIRED)
+
 # Workaround for https://bugreports.qt-project.org/browse/QTBUG-29987
 set(QT_IMPORTS_DIR "${CMAKE_INSTALL_LIBDIR}/qt5/qml")
 set(CMAKE_AUTOMOC ON)
diff --git a/lib/SystemSettings/CMakeLists.txt b/lib/SystemSettings/CMakeLists.txt
index 1c3245b4..889ad3a7 100644
--- a/lib/SystemSettings/CMakeLists.txt
+++ b/lib/SystemSettings/CMakeLists.txt
@@ -4,7 +4,7 @@ VERSION 1.0.0
 SOVERSION 1
 )
 
-qt5_use_modules(SystemSettings Core Gui Quick Qml)
+target_link_libraries(SystemSettings Qt5::Core Qt5::Gui Qt5::Quick Qt5::Qml)
 
 install(TARGETS SystemSettings LIBRARY DESTINATION ${LIBDIR})
 install(FILES item-base.h ItemBase plugin-interface.h PluginInterface
diff --git a/plugins/background/CMakeLists.txt b/plugins/background/CMakeLists.txt
index ddeea901..77c33936 100644
--- a/plugins/background/CMakeLists.txt
+++ b/plugins/background/CMakeLists.txt
@@ -9,9 +9,8 @@ set(QML_SOURCES
 add_library(UbuntuBackgroundPanel MODULE
 plugin.cpp background.cpp plugin.h background.h
 ${QML_SOURCES}) # So they show up in Qt designer.
-qt5_use_modules(UbuntuBackgroundPanel Qml Quick DBus)
 
-target_link_libraries(UbuntuBackgroundPanel uss-accountsservice)
+target_link_libraries(UbuntuBackgroundPanel Qt5::Qml Qt5::Quick Qt5::DBus uss-accountsservice)
 
 set(PLUG_DIR ${PLUGIN_PRIVATE_MODULE_DIR}/Ubuntu/SystemSettings/Background)
 install(TARGETS UbuntuBackgroundPanel DESTINATION ${PLUG_DIR})
diff --git a/plugins/battery/CMakeLists.txt b/plugins/battery/CMakeLists.txt
index 611a0043..2f05ee42 100644
--- a/plugins/battery/CMakeLists.txt
+++ b/plugins/battery/CMakeLists.txt
@@ -10,8 +10,7 @@ set(QML_SOURCES
 include_directories(${GLIB_INCLUDE_DIRS} ${UPOWER_GLIB_INCLUDE_DIRS})
 add_library(UbuntuBatteryPanel MODULE plugin.h battery.h plugin.cpp battery.cpp
 ${QML_SOURCES})
-qt5_use_modules(UbuntuBatteryPanel Quick Qml DBus)
-target_link_libraries(UbuntuBatteryPanel ${GLIB_LDFLAGS} ${UPOWER_GLIB_LDFLAGS})
+target_link_libraries(UbuntuBatteryPanel Qt5::Quick Qt5::Qml Qt5::DBus ${GLIB_LDFLAGS} ${UPOWER_GLIB_LDFLAGS})
 
 set(PLUG_DIR ${PLUGIN_PRIVATE_MODULE_DIR}/Ubuntu/SystemSettings/Battery)
 install(TARGETS UbuntuBatteryPanel DESTINATION ${PLUG_DIR})
diff --git a/plugins/battery/plugin/CMakeLists.txt b/plugins/battery/plugin/CMakeLists.txt
index 342b31d5..5d1c874f 100644
--- a/plugins/battery/plugin/CMakeLists.txt
+++ b/plugins/battery/plugin/CMakeLists.txt
@@ -6,6 +6,5 @@ add_definitions(-DQT_NO_KEYWORDS)
 
 
 add_library(battery-plugin SHARED battery-plugin.h battery-plugin.cpp)
-qt5_use_modules(battery-plugin Core Qml)
-target_link_libraries(battery-plugin SystemSettings ${GLIB_LDFLAGS} ${UPOWER_GLIB_LDFLAGS})
+target_link_libraries(battery-plugin Qt5::Core Qt5::Qml SystemSettings ${GLIB_LDFLAGS} ${UPOWER_GLIB_LDFLAGS})
 install(TARGETS battery-plugin DESTINATION ${PLUGIN_MODULE_DIR})
diff --git a/plugins/bluetooth/CMakeLists.txt b/plugins/bluetooth/CMakeLists.txt
index 7109fce0..3b46d7aa 100644
--- a/plugins/bluetooth/CMakeLists.txt
+++ b/plugins/bluetooth/CMakeLists.txt
@@ -38,7 +38,7 @@ add_library(UbuntuBluetoothPanel MODULE
   ${QML_SOURCES}
 )
 
-qt5_use_modules(UbuntuBluetoothPanel Qml Quick DBus)
+target_link_libraries(UbuntuBluetoothPanel Qt5::Qml Qt5::Quick Qt5::DBus)
 
 set(PLUG_DIR ${PLUGIN_PRIVATE_MODULE_DIR}/Ubuntu/SystemSettings/Bluetooth)
 install(TARGETS UbuntuBluetoothPanel DESTINATION ${PLUG_DIR})
diff --git a/plugins/cellular/CMakeLists.txt b/plugins/cellular/CMakeLists.txt
index c0441fce..2b9c360e 100644
--- a/plugins/cellular/CMakeLists.txt
+++ b/plugins/cellular/CMakeLists.txt
@@ -30,8 +30,7 @@ add_library(UbuntuCellularPanel MODULE
   nm_settings_connection_proxy.h
   ${QML_SOURCES}
 )
-target_link_libraries(UbuntuCellularPanel uss-accountsservice)
-qt5_use_modules(UbuntuCellularPanel Qml Quick DBus)
+target_link_libraries(UbuntuCellularPanel uss-accountsservice Qt5::Qml Qt5::Quick Qt5::DBus)
 
 set(PLUG_DIR ${PLUGIN_PRIVATE_MODULE_DIR}/Ubuntu/SystemSettings/Cellular)
 install(TARGETS UbuntuCellularPanel DESTINATION ${PLUG_DIR})
diff --git a/plugins/cellular/plugin/CMakeLists.txt b/plugins/cellular/plugin/CMakeLists.txt
index d7ada601..5a2bb4a6 100644
--- a/plugins/cellular/plugin/CMakeLists.txt
+++ b/plugins/cellular/plugin/CMakeLists.txt
@@ -1,6 +1,5 @@
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 add_library(cellular-plugin SHARED cellular-plugin.h cellular-plugin.cpp)
-qt5_use_modules(cellular-plugin Core Qml DBus)
-target_link_libraries(cellular-plugin SystemSettings)
+target_link_libraries(cellular-plugin Qt5::Core Qt5::Qml Qt5::DBus SystemSettings)
 install(TARGETS cellular-plugin DESTINATION ${PLUGIN_MODULE_DIR})
diff --git a/plugins/example/CMakeLists.txt b/plugins/example/CMakeLists.txt
index 75271a97..fbdbac7b 100644
--- a/plugins/example/CMakeLists.txt
+++ b/plugins/example/CMakeLists.txt
@@ -2,8 +2,7 @@ include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 set(QML_SOURCES PageComponent.qml)
 add_library(example-plugin MODULE example-plugin.cpp example-plugin.h ${QML_SOURCES})
-qt5_use_modules(example-plugin Qml Core)
-target_link_libraries(example-plugin SystemSettings)
+target_link_libraries(example-plugin Qt5::Qml Qt5::Core SystemSettings)
 
 # This plugin is only an example so let's not install it.
 #install(FILES example.settings DESTINATION ${PLUGIN_MANIFEST_DIR})
diff --git a/plugins/flight-mode/CMakeLists.txt b/plugins/flight-mode/CMakeLists.txt
index 95478c9a..c563c14e 100644
--- a/plugins/flight-mode/CMakeLists.txt
+++ b/plugins/flight-mode/CMakeLists.txt
@@ -12,7 +12,7 @@ add_library(FlightModeHelper MODULE
   urfkill-proxy.h
   ${QML_SOURCES}
 )
-qt5_use_modules(FlightModeHelper Qml Quick DBus)
+target_link_libraries(FlightModeHelper Qt5::Qml Qt5::Quick Qt5::DBus)
 set(PLUG_DIR ${PLUGIN_PRIVATE_MODULE_DIR}/Ubuntu/SystemSettings/FlightMode)
 
 install(FILES ${QML_SOURCES} DESTINATION ${PLUGIN_QML_DIR}/flight-mode)
diff --git a/plugins/hotspot/plugin/CMakeLists.txt b/plugins/hotspot/plugin/CMakeLists.txt
index 87e19372..7c1b6cf3 100644
--- a/plugins/hotspot/plugin/CMakeLists.txt
+++ b/plugins/hotspot/plugin/CMakeLists.txt
@@ -3,6 +3,5 @@ include_directories(${CMAKE_CURRENT_BINARY_DIR})
 add_definitions(-DQT_NO_KEYWORDS)
 
 add_library(hotspot-plugin SHARED hotspot-plugin.h hotspot-plugin.cpp)
-qt5_use_modules(hotspot-plugin Core Qml DBus)
-target_link_libraries(hotspot-plugin SystemSettings)
+target_link_libraries(hotspot-plugin Qt5::Core Qt5::Qml Qt5::DBus SystemSettings)
 install(TARGETS hotspot-plugin DESTINATION ${PLUGIN_MODULE_DIR})
diff --git a/plugins/language/CMakeLists.txt b/plugins/language/CMakeLists.txt
index e00fb8e5..fcca4633 100644
--- a/plugins/language/CMakeLists.txt
+++ b/plugins/language/CMakeLists.txt
@@ -18,8 +18,7 @@ add_library(UbuntuLanguagePlugin MODULE
  keyboard-layout.cpp language-plugin.cpp plugin.cpp subset-model.cpp onscreenkeyboard-plugin.cpp hardwarekeyboard-plugin.cpp
  keyboard-layout.h language-plugin.h plugin.h subset-model.h onscreenkeyboard-plugin.h hardwarekeyboard-plugin.h
  ${QML_SOURCES})
-qt5_use_modules(UbuntuLanguagePlugin Qml Quick DBus)
-target_link_libraries(UbuntuLanguagePlugin uss-accountsservice uss-sessionservice ${GD3_LDFLAGS} ${GLIB_LDFLAGS} ${GIO_LDFLAGS} ${ACCOUNTSSERVICE_LDFLAGS} ${ICU_LDFLAGS})
+target_link_libraries(UbuntuLanguagePlugin Qt5::Qml Qt5::Quick Qt5::DBus uss-accountsservice uss-sessionservice ${GD3_LDFLAGS} ${GLIB_LDFLAGS} ${GIO_LDFLAGS} ${ACCOUNTSSERVICE_LDFLAGS} ${ICU_LDFLAGS})
 
 set(PLUG_DIR ${PLUGIN_PRIVATE_MODULE_DIR}/Ubuntu/SystemSettings/LanguagePlugin)
 install(TARGETS UbuntuLanguagePlugin DESTINATION ${PLUG_DIR})
diff --git a/plugins/launcher/CMakeLists.txt b/plugins/launcher/CMakeLists.txt
index f8ee31f6..3f360a7b 100644
--- a/plugins/launcher/CMakeLists.txt
+++ b/plugins/launcher/CMakeLists.txt
@@ -10,7 +10,7 @@ set(PANEL_SOURCES
 )
 
 add_library(UbuntuLauncherPanel MODULE ${PANEL_SOURCES} ${QML_SOURCES})
-qt5_use_modules(UbuntuLauncherPanel Qml Widgets)
+target_link_libraries(UbuntuLauncherPanel Qt5::Qml Qt5::Widgets)
 
 set(PLUG_DIR ${PLUGIN_PRIVATE_MODULE_DIR}/Ubuntu/SystemSettings/Launcher)
 install(TARGETS UbuntuLauncherPanel DESTINATION ${PLUG_DIR})
diff --git a/plugins/launcher/plugin/CMakeLists.txt b/plugins/launcher/plugin/CMakeLists.txt
index 58b8fc39..3b99f12e 100644
--- a/plugins/launcher/plugin/CMakeLists.txt
+++ b/plugins/launcher/plugin/CMakeLists.txt
@@ -2,7 +2,6 @@ include_directories(${CMAKE_CURRENT_BINARY_DIR} ../)
 
 add_definitions(-DLAUNCHER_PLUGIN_QML_DIR="${PLUGIN_QML_DIR}/launcher")
 add_library(launcher-plugin SHARED launcher-plugin.h launcher-plugin.cpp ../launcher.h ../launcher_impl.cpp)
-qt5_use_modules(launcher-plugin Core Qml Widgets)
-target_link_libraries(launcher-plugin SystemSettings)
+target_link_libraries(launcher-plugin Qt5::Core Qt5::Qml Qt5::Widgets SystemSettings)
 install(TARGETS launcher-plugin DESTINATION ${PLUGIN_MODULE_DIR})
 install(FILES GuAccessor.qml DESTINATION ${PLUGIN_QML_DIR}/launcher)
diff --git a/plugins/mouse/CMakeLists.txt b/plugins/mouse/CMakeLists.txt
index a8a16d45..8df9da8d 100644
--- a/plugins/mouse/CMakeLists.txt
+++ b/plugins/mouse/CMakeLists.txt
@@ -18,7 +18,6 @@ set(PANEL_SOURCES
 
 add_definitions(-DQT_NO_KEYWORDS)
 add_library(UbuntuMousePanel MODULE ${PANEL_SOURCES} ${QML_SOURCES})
-qt5_use_modules(UbuntuMousePanel Qml Quick DBus)
 
 include_directories(${CMAKE_CURRENT_BINARY_DIR}
                     ${ACCOUNTSSERVICE_INCLUDE_DIRS}
@@ -26,6 +25,7 @@ include_directories(${CMAKE_CURRENT_BINARY_DIR}
 
 set(PLUG_DIR ${PLUGIN_PRIVATE_MODULE_DIR}/Ubuntu/SystemSettings/Mouse)
 target_link_libraries (UbuntuMousePanel
+    Qt5::Qml Qt5::Quick Qt5::DBus
     uss-accountsservice
     ${ACCOUNTSSERVICE_LDFLAGS}
     ${GOBJECT_LDFLAGS}
diff --git a/plugins/notifications/CMakeLists.txt b/plugins/notifications/CMakeLists.txt
index 932c695b..8c236693 100644
--- a/plugins/notifications/CMakeLists.txt
+++ b/plugins/notifications/CMakeLists.txt
@@ -29,8 +29,7 @@ add_definitions(-DQT_NO_KEYWORDS)
 include_directories(${GIO_INCLUDE_DIRS})
 include_directories(${QTGSETTINGS_INCLUDE_DIRS})
 
-target_link_libraries(UbuntuNotificationsPanel ${GIO_LDFLAGS} ${QTGSETTINGS_LDFLAGS})
-qt5_use_modules(UbuntuNotificationsPanel Qml Quick)
+target_link_libraries(UbuntuNotificationsPanel ${GIO_LDFLAGS} ${QTGSETTINGS_LDFLAGS} Qt5::Qml Qt5::Quick)
 
 set(PLUG_DIR ${PLUGIN_PRIVATE_MODULE_DIR}/Ubuntu/SystemSettings/Notifications)
 install(TARGETS UbuntuNotificationsPanel DESTINATION ${PLUG_DIR})
diff --git a/plugins/notifications/plugin/CMakeLists.txt b/plugins/notifications/plugin/CMakeLists.txt
index 6c965672..0bf0df5b 100644
--- a/plugins/notifications/plugin/CMakeLists.txt
+++ b/plugins/notifications/plugin/CMakeLists.txt
@@ -1,6 +1,5 @@
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 add_library(notifications-plugin SHARED notifications-plugin.h notifications-plugin.cpp)
-qt5_use_modules(notifications-plugin Core Qml)
-target_link_libraries(notifications-plugin SystemSettings)
+target_link_libraries(notifications-plugin Qt5::Core Qt5::Qml SystemSettings)
 install(TARGETS notifications-plugin DESTINATION ${PLUGIN_MODULE_DIR})
diff --git a/plugins/phone/plugin/CMakeLists.txt b/plugins/phone/plugin/CMakeLists.txt
index 48b99db3..35b39d94 100644
--- a/plugins/phone/plugin/CMakeLists.txt
+++ b/plugins/phone/plugin/CMakeLists.txt
@@ -1,6 +1,5 @@
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 add_library(phone-plugin SHARED phone-plugin.h phone-plugin.cpp)
-qt5_use_modules(phone-plugin Core Qml DBus)
-target_link_libraries(phone-plugin SystemSettings)
+target_link_libraries(phone-plugin Qt5::Core Qt5::Qml Qt5::DBus SystemSettings)
 install(TARGETS phone-plugin DESTINATION ${PLUGIN_MODULE_DIR})
diff --git a/plugins/reset/CMakeLists.txt b/plugins/reset/CMakeLists.txt
index 69bf1359..e6bf6d95 100644
--- a/plugins/reset/CMakeLists.txt
+++ b/plugins/reset/CMakeLists.txt
@@ -12,9 +12,8 @@ add_library(UbuntuResetPanel MODULE
   reset.h
   ${QML_SOURCES}
 )
-qt5_use_modules(UbuntuResetPanel Qml Quick DBus)
 
-target_link_libraries(UbuntuResetPanel uss-accountsservice)
+target_link_libraries(UbuntuResetPanel Qt5::Qml Qt5::Quick Qt5::DBus uss-accountsservice)
 
 set(PLUG_DIR ${PLUGIN_PRIVATE_MODULE_DIR}/Ubuntu/SystemSettings/Reset)
 install(TARGETS UbuntuResetPanel DESTINATION ${PLUG_DIR})
diff --git a/plugins/security-privacy/CMakeLists.txt b/plugins/security-privacy/CMakeLists.txt
index 823aefc6..a1dbb2cc 100644
--- a/plugins/security-privacy/CMakeLists.txt
+++ b/plugins/security-privacy/CMakeLists.txt
@@ -44,7 +44,6 @@ set(PANEL_SOURCES
 )
 
 add_library(UbuntuSecurityPrivacyPanel MODULE ${PANEL_SOURCES} ${QML_SOURCES})
-qt5_use_modules(UbuntuSecurityPrivacyPanel Qml Quick DBus)
 
 pkg_check_modules(TRUST_STORE REQUIRED trust-store)
 
@@ -56,6 +55,7 @@ include_directories(
 
 set(PLUG_DIR ${PLUGIN_PRIVATE_MODULE_DIR}/Ubuntu/SystemSettings/SecurityPrivacy)
 target_link_libraries (UbuntuSecurityPrivacyPanel
+    Qt5::Qml Qt5::Quick Qt5::DBus
     uss-accountsservice
     ${ACCOUNTSSERVICE_LDFLAGS}
     ${GOBJECT_LDFLAGS}
diff --git a/plugins/security-privacy/diagnostics/CMakeLists.txt b/plugins/security-privacy/diagnostics/CMakeLists.txt
index ca17b0d8..3c2b63d4 100644
--- a/plugins/security-privacy/diagnostics/CMakeLists.txt
+++ b/plugins/security-privacy/diagnostics/CMakeLists.txt
@@ -5,7 +5,7 @@ set(QML_SOURCES
 
 add_library(UbuntuDiagnostics MODULE plugin.h diagnostics.h plugin.cpp diagnostics.cpp
 ${QML_SOURCES})
-qt5_use_modules(UbuntuDiagnostics Qml Quick DBus)
+target_link_libraries(UbuntuDiagnostics Qt5::Qml Qt5::Quick Qt5::DBus)
 
 set(PLUG_DIR ${PLUGIN_PRIVATE_MODULE_DIR}/Ubuntu/SystemSettings/Diagnostics)
 install(TARGETS UbuntuDiagnostics DESTINATION ${PLUG_DIR})
diff --git a/plugins/sound/CMakeLists.txt b/plugins/sound/CMakeLists.txt
index c698a4f9..0e2c6ff6 100644
--- a/plugins/sound/CMakeLists.txt
+++ b/plugins/sound/CMakeLists.txt
@@ -5,8 +5,7 @@ set(QML_SOURCES
 
 add_library(UbuntuSoundPanel MODULE plugin.h sound.h plugin.cpp sound.cpp
 ${QML_SOURCES})
-target_link_libraries(UbuntuSoundPanel uss-accountsservice)
-qt5_use_modules(UbuntuSoundPanel Qml Quick DBus)
+target_link_libraries(UbuntuSoundPanel uss-accountsservice Qt5::Qml Qt5::Quick Qt5::DBus)
 
 set(PLUG_DIR ${PLUGIN_PRIVATE_MODULE_DIR}/Ubuntu/SystemSettings/Sound)
 install(TARGETS UbuntuSoundPanel DESTINATION ${PLUG_DIR})
diff --git a/plugins/system-update/CMakeLists.txt b/plugins/system-update/CMakeLists.txt
index 881c9fb0..948376f6 100644
--- a/plugins/system-update/CMakeLists.txt
+++ b/plugins/system-update/CMakeLists.txt
@@ -44,8 +44,8 @@ add_library(UpdatePlugin SHARED
     updatemanager.cpp
     ../../src/i18n.cpp
 )
-qt5_use_modules(UpdatePlugin Quick Core Network Sql)
 target_link_libraries(UpdatePlugin
+    Qt5::Quick Qt5::Core Qt5::Network Qt5::Sql
     apt-pkg
     uss-systemimage
     ${UAL_LDFLAGS}
@@ -55,8 +55,7 @@ add_library(UbuntuUpdatePanel MODULE
     plugin.cpp
     ${QML_SOURCES}
 )
-qt5_use_modules(UbuntuUpdatePanel Qml Quick DBus)
-target_link_libraries(UbuntuUpdatePanel UpdatePlugin)
+target_link_libraries(UbuntuUpdatePanel Qt5::Qml Qt5::Quick Qt5::DBus UpdatePlugin)
 
 set(PLUG_DIR ${PLUGIN_PRIVATE_MODULE_DIR}/Ubuntu/SystemSettings/Update)
 install(TARGETS UpdatePlugin DESTINATION ${PLUGIN_MODULE_DIR})
diff --git a/plugins/time-date/CMakeLists.txt b/plugins/time-date/CMakeLists.txt
index 3fffbf81..9ebc0fc8 100644
--- a/plugins/time-date/CMakeLists.txt
+++ b/plugins/time-date/CMakeLists.txt
@@ -12,8 +12,7 @@ add_library(UbuntuTimeDatePanel MODULE plugin.h
   timezonelocationmodel.cpp
   ${QML_SOURCES}
 )
-qt5_use_modules(UbuntuTimeDatePanel Qml Quick DBus Concurrent)
-target_link_libraries(UbuntuTimeDatePanel ${GLIB_LDFLAGS} ${GEONAMES_LDFLAGS})
+target_link_libraries(UbuntuTimeDatePanel Qt5::Qml Qt5::Quick Qt5::DBus Qt5::Concurrent ${GLIB_LDFLAGS} ${GEONAMES_LDFLAGS})
 
 
 set(PLUG_DIR ${PLUGIN_PRIVATE_MODULE_DIR}/Ubuntu/SystemSettings/TimeDate)
diff --git a/plugins/wifi/CMakeLists.txt b/plugins/wifi/CMakeLists.txt
index fc850392..5aa16972 100644
--- a/plugins/wifi/CMakeLists.txt
+++ b/plugins/wifi/CMakeLists.txt
@@ -36,7 +36,7 @@ add_library(UbuntuWifiPanel MODULE
   wifidbushelper.h
   ${QML_SOURCES}
 )
-qt5_use_modules(UbuntuWifiPanel Qml Quick DBus)
+target_link_libraries(UbuntuWifiPanel Qt5::Qml Qt5::Quick Qt5::DBus)
 
 set(PLUG_DIR ${PLUGIN_PRIVATE_MODULE_DIR}/Ubuntu/SystemSettings/Wifi)
 install(TARGETS UbuntuWifiPanel DESTINATION ${PLUG_DIR})
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index d40e5ea3..5cc4ee32 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -34,22 +34,21 @@ set(QML_SOURCES
 QT5_ADD_RESOURCES(system-settings-resources ui.qrc)
 
 add_executable(system-settings ${USS_SOURCES} ${QML_SOURCES} ${system-settings-resources})
-qt5_use_modules(system-settings Core Gui Quick Qml DBus Widgets)
-target_link_libraries(system-settings SystemSettings ${GLIB_LDFLAGS})
+target_link_libraries(system-settings Qt5::Core Qt5::Gui Qt5::Quick Qt5::Qml Qt5::DBus Qt5::Widgets SystemSettings ${GLIB_LDFLAGS})
 install(TARGETS system-settings RUNTIME DESTINATION bin)
 
 add_library(uss-accountsservice SHARED accountsservice.h accountsservice.cpp)
-qt5_use_modules(uss-accountsservice Core Qml DBus)
+target_link_libraries(uss-accountsservice Qt5::Core Qt5::Qml Qt5::DBus)
 set_target_properties(uss-accountsservice PROPERTIES VERSION 0.0 SOVERSION 0.0)
 install(TARGETS uss-accountsservice LIBRARY DESTINATION ${PLUGIN_MODULE_DIR} NAMELINK_SKIP)
 
 add_library(uss-sessionservice SHARED sessionservice.h sessionservice.cpp)
-qt5_use_modules(uss-sessionservice Core Qml DBus)
+target_link_libraries(uss-sessionservice Qt5::Core Qt5::Qml Qt5::DBus)
 set_target_properties(uss-sessionservice PROPERTIES VERSION 0.0 SOVERSION 0.0)
 install(TARGETS uss-sessionservice LIBRARY DESTINATION ${PLUGIN_MODULE_DIR} NAMELINK_SKIP)
 
 add_library(uss-systemimage SHARED systemimage.h systemimage.cpp i18n.cpp)
-qt5_use_modules(uss-systemimage Core Qml DBus)
+target_link_libraries(uss-systemimage Qt5::Core Qt5::Qml Qt5::DBus)
 set_target_properties(uss-systemimage PROPERTIES VERSION 0.0 SOVERSION 0.0)
 install(TARGETS uss-systemimage LIBRARY DESTINATION ${PLUGIN_MODULE_DIR} NAMELINK_SKIP)
 
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 058a108f..40ebf3d8 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -20,12 +20,10 @@ add_definitions(-DSYSTEM_IMAGE_DBUS_TEMPLATE="${CMAKE_SOURCE_DIR}/tests/autopilo
 add_definitions(-DI18N_DOMAIN="ubuntu-system-settings")
 
 add_library(test-plugin SHARED test-plugin.cpp test-plugin.h)
-target_link_librarieS(test-plugin SystemSettings)
-qt5_use_modules(test-plugin Core Qml)
+target_link_librarieS(test-plugin SystemSettings Qt5::Core Qt5::Qml)
 
 add_library(test-plugin2 SHARED test-plugin2.cpp test-plugin2.h)
-target_link_librarieS(test-plugin2 SystemSettings)
-qt5_use_modules(test-plugin2 Core Qml)
+target_link_librarieS(test-plugin2 SystemSettings Qt5::Core Qt5::Qml)
 
 add_executable(tst-plugins
     tst_plugins.cpp
@@ -44,24 +42,22 @@ add_executable(tst-arguments
     ../src/utils.cpp
 )
 
-qt5_use_modules(tst-plugins Core Qml Test)
-target_link_libraries(tst-plugins SystemSettings ${GLIB_LDFLAGS})
+target_link_libraries(tst-plugins Qt5::Core Qt5::Qml Qt5::Test SystemSettings ${GLIB_LDFLAGS})
 add_test(tst-plugins tst-plugins)
 set_tests_properties(tst-plugins PROPERTIES ENVIRONMENT
     "QT_QPA_PLATFORM=minimal;XDG_DATA_DIRS=${CMAKE_CURRENT_SOURCE_DIR}"
 )
 
-qt5_use_modules(tst-arguments Core Test)
-target_link_libraries(tst-arguments ${GLIB_LDFLAGS})
+target_link_libraries(tst-arguments Qt5::Core Qt5::Test ${GLIB_LDFLAGS})
 add_test(tst-arguments tst-arguments)
 
 add_executable(tst-systemimage
     tst_systemimage.cpp
     mocks/system-image-dbus/fakesystemimagedbus.cpp
 )
-qt5_use_modules(tst-systemimage Core DBus Test)
 add_test(tst-systemimage tst-systemimage)
 target_link_libraries(tst-systemimage
+    Qt5::Core Qt5::DBus Qt5::Test
     uss-systemimage
     ${QTDBUSMOCK_LIBRARIES}
     ${QTDBUSTEST_LIBRARIES}
diff --git a/tests/mocks/GSettings.1.0/CMakeLists.txt b/tests/mocks/GSettings.1.0/CMakeLists.txt
index 4e6eb637..88fb6d60 100644
--- a/tests/mocks/GSettings.1.0/CMakeLists.txt
+++ b/tests/mocks/GSettings.1.0/CMakeLists.txt
@@ -13,7 +13,7 @@ add_library(FakeGSettingsQml MODULE ${GSettingsQML_SOURCES})
 target_link_libraries(FakeGSettingsQml
   ${Qt5Core_LIBRARIES}
   ${Qt5Quick_LIBRARIES}
+  Qt5::Qml
 )
 
-qt5_use_modules(FakeGSettingsQml Qml)
 add_uss_mock(GSettings 1.0 GSettings.1.0 TARGETS FakeGSettingsQml)
diff --git a/tests/mocks/QMenuModel/CMakeLists.txt b/tests/mocks/QMenuModel/CMakeLists.txt
index a108c23a..125d37a0 100644
--- a/tests/mocks/QMenuModel/CMakeLists.txt
+++ b/tests/mocks/QMenuModel/CMakeLists.txt
@@ -14,10 +14,9 @@ add_library(QMenuModelQml SHARED ${QMenuModelQml_SOURCES})
 target_link_libraries(QMenuModelQml
   ${Qt5Core_LIBRARIES}
   ${Qt5Quick_LIBRARIES}
+  Qt5::Core Qt5::Quick
 )
 
-qt5_use_modules(QMenuModelQml Core Quick)
-
 set_target_properties(QMenuModelQml PROPERTIES
                       OUTPUT_NAME qmenumodel
                       VERSION 0)
diff --git a/tests/mocks/Ubuntu/Connectivity/CMakeLists.txt b/tests/mocks/Ubuntu/Connectivity/CMakeLists.txt
index 967561b4..f5d6bd02 100644
--- a/tests/mocks/Ubuntu/Connectivity/CMakeLists.txt
+++ b/tests/mocks/Ubuntu/Connectivity/CMakeLists.txt
@@ -9,7 +9,7 @@ set(MOCK_CONNECTIVITY_SRCS
 
 add_library(MockConnectivity MODULE ${MOCK_CONNECTIVITY_SRCS})
 
-qt5_use_modules(MockConnectivity Qml Quick Core)
+target_link_libraries(MockConnectivity Qt5::Qml Qt5::Quick Qt5::Core)
 
 add_uss_mock(Ubuntu.Conectivity 1.0 Ubuntu/Connectivity
              TARGETS MockConnectivity)
diff --git a/tests/mocks/Ubuntu/DownloadManager/CMakeLists.txt b/tests/mocks/Ubuntu/DownloadManager/CMakeLists.txt
index 60a16fc4..f89adf4a 100644
--- a/tests/mocks/Ubuntu/DownloadManager/CMakeLists.txt
+++ b/tests/mocks/Ubuntu/DownloadManager/CMakeLists.txt
@@ -11,7 +11,7 @@ set(MOCK_DOWNLOADMANAGER_SRCS
 
 add_library(MockUbuntuDownloadManager MODULE ${MOCK_DOWNLOADMANAGER_SRCS})
 
-qt5_use_modules(MockUbuntuDownloadManager Qml Quick Core)
+target_link_libraries(MockUbuntuDownloadManager Qt5::Qml Qt5::Quick Qt5::Core)
 
 add_uss_mock(Ubuntu.DownloadManager 1.2 Ubuntu/DownloadManager
              TARGETS MockUbuntuDownloadManager)
diff --git a/tests/mocks/Ubuntu/OnlineAccounts/Client/CMakeLists.txt b/tests/mocks/Ubuntu/OnlineAccounts/Client/CMakeLists.txt
index 77f04882..ff5647dd 100644
--- a/tests/mocks/Ubuntu/OnlineAccounts/Client/CMakeLists.txt
+++ b/tests/mocks/Ubuntu/OnlineAccounts/Client/CMakeLists.txt
@@ -12,8 +12,8 @@ add_library(OnlineAccountsClientQml SHARED ${OnlineAccountsClientQml_SRC})
 target_link_libraries(OnlineAccountsClientQml
   ${Qt5Core_LIBRARIES}
   ${Qt5Quick_LIBRARIES}
+  Qt5::Core Qt5::Quick
 )
 
-qt5_use_modules(OnlineAccountsClientQml Core Quick)
 add_uss_mock(Ubuntu.OnlineAccounts.Client 0.1 Ubuntu/OnlineAccounts/Client
              TARGETS OnlineAccountsClientQml)
diff --git a/tests/mocks/Ubuntu/SystemSettings/Notifications/CMakeLists.txt b/tests/mocks/Ubuntu/SystemSettings/Notifications/CMakeLists.txt
index 6865176d..83f08931 100644
--- a/tests/mocks/Ubuntu/SystemSettings/Notifications/CMakeLists.txt
+++ b/tests/mocks/Ubuntu/SystemSettings/Notifications/CMakeLists.txt
@@ -15,8 +15,7 @@ set(MOCK_NOTIFICATIONS_SRCS
 
 add_library(MockUbuntuNotificationsPanel MODULE ${MOCK_NOTIFICATIONS_SRCS})
 
-target_link_libraries(MockUbuntuNotificationsPanel ${GIO_LDFLAGS} ${QTGSETTINGS_LDFLAGS})
-qt5_use_modules(MockUbuntuNotificationsPanel Qml Quick Core)
+target_link_libraries(MockUbuntuNotificationsPanel ${GIO_LDFLAGS} ${QTGSETTINGS_LDFLAGS} Qt5::Qml Qt5::Quick Qt5::Core)
 
 add_uss_mock(Ubuntu.SystemSettings.Notifications 1.0 Ubuntu/SystemSettings/Notifications
              TARGETS MockUbuntuNotificationsPanel)
diff --git a/tests/mocks/Ubuntu/SystemSettings/Update/CMakeLists.txt b/tests/mocks/Ubuntu/SystemSettings/Update/CMakeLists.txt
index be31ab62..373dfaf4 100644
--- a/tests/mocks/Ubuntu/SystemSettings/Update/CMakeLists.txt
+++ b/tests/mocks/Ubuntu/SystemSettings/Update/CMakeLists.txt
@@ -21,8 +21,7 @@ set(MOCK_SYSTEMUPDATE_SRCS
 )
 
 add_library(MockSystemUpdate MODULE ${MOCK_SYSTEMUPDATE_SRCS})
-target_link_libraries(MockSystemUpdate ${PLUGIN_LIBS})
-qt5_use_modules(MockSystemUpdate Qml Quick Core Sql)
+target_link_libraries(MockSystemUpdate ${PLUGIN_LIBS} Qt5::Qml Qt5::Quick Qt5::Core Qt5::Sql)
 
 add_uss_mock(Ubuntu.SystemSettings.Update 1.0 Ubuntu/SystemSettings/Update
              TARGETS MockSystemUpdate)
diff --git a/tests/plugins/bluetooth/CMakeLists.txt b/tests/plugins/bluetooth/CMakeLists.txt
index 3231a8ba..5c37d7d4 100644
--- a/tests/plugins/bluetooth/CMakeLists.txt
+++ b/tests/plugins/bluetooth/CMakeLists.txt
@@ -26,6 +26,7 @@ add_executable(tst-bluetooth
 target_link_libraries(tst-bluetooth
     ${QTDBUSMOCK_LIBRARIES}
     ${QTDBUSTEST_LIBRARIES}
+    Qt5::Qml Qt5::Quick Qt5::Core Qt5::DBus Qt5::Test
 )
 
 add_executable(tst-bluetooth-devicemodel
@@ -37,6 +38,7 @@ add_executable(tst-bluetooth-devicemodel
 target_link_libraries(tst-bluetooth-devicemodel
     ${QTDBUSMOCK_LIBRARIES}
     ${QTDBUSTEST_LIBRARIES}
+    Qt5::Qml Qt5::Quick Qt5::Core Qt5::DBus Qt5::Test
 )
 
 add_executable(tst-bluetooth-device
@@ -48,12 +50,9 @@ add_executable(tst-bluetooth-device
 target_link_libraries(tst-bluetooth-device
     ${QTDBUSMOCK_LIBRARIES}
     ${QTDBUSTEST_LIBRARIES}
+    Qt5::Qml Qt5::Quick Qt5::Core Qt5::DBus Qt5::Test
 )
 
-qt5_use_modules(tst-bluetooth Qml Quick Core DBus Test)
-qt5_use_modules(tst-bluetooth-devicemodel Qml Quick Core DBus Test)
-qt5_use_modules(tst-bluetooth-device Qml Quick Core DBus Test)
-
 add_test(NAME tst-bluetooth
          COMMAND ${XVFB_CMD} ${CMAKE_CURRENT_BINARY_DIR}/tst-bluetooth)
 add_test(NAME tst-bluetooth-devicemodel
diff --git a/tests/plugins/security-privacy/CMakeLists.txt b/tests/plugins/security-privacy/CMakeLists.txt
index 26344c0e..e1bae37f 100644
--- a/tests/plugins/security-privacy/CMakeLists.txt
+++ b/tests/plugins/security-privacy/CMakeLists.txt
@@ -7,6 +7,5 @@ add_executable(tst-trust-store-model
     tst_trust_store_model.cpp
     ../../../plugins/security-privacy/trust-store-model.cpp
 )
-target_link_libraries (tst-trust-store-model ${GLIB_LDFLAGS})
-qt5_use_modules(tst-trust-store-model Core Gui DBus Qml Test)
+target_link_libraries (tst-trust-store-model ${GLIB_LDFLAGS} Qt5::Core Qt5::Gui Qt5::DBus Qt5::Qml Qt5::Test)
 add_test(NAME tst-trust-store-model COMMAND ${XVFB_CMD} ${CMAKE_CURRENT_BINARY_DIR}/tst-trust-store-model)
diff --git a/tests/plugins/wifi/CMakeLists.txt b/tests/plugins/wifi/CMakeLists.txt
index b60dc7a8..a2d9e04c 100644
--- a/tests/plugins/wifi/CMakeLists.txt
+++ b/tests/plugins/wifi/CMakeLists.txt
@@ -18,6 +18,5 @@ add_executable(tst-wifidbushelper
     ${CMAKE_SOURCE_DIR}/plugins/wifi/wifidbushelper.cpp
     ${CMAKE_SOURCE_DIR}/tests/mocks/plugins/wifi/fakenetworkmanager.cpp
 )
-qt5_use_modules(tst-wifidbushelper Core DBus Network Test)
-target_link_libraries(tst-wifidbushelper ${QTDBUSMOCK_LIBRARIES} ${QTDBUSTEST_LIBRARIES})
+target_link_libraries(tst-wifidbushelper Qt5::Core Qt5::DBus Qt5::Network Qt5::Test ${QTDBUSMOCK_LIBRARIES} ${QTDBUSTEST_LIBRARIES})
 add_test(tst-wifidbushelper tst-wifidbushelper)
diff --git a/tests/utils/modules/Ubuntu/Test/CMakeLists.txt b/tests/utils/modules/Ubuntu/Test/CMakeLists.txt
index 1c7b9b68..b4199c11 100644
--- a/tests/utils/modules/Ubuntu/Test/CMakeLists.txt
+++ b/tests/utils/modules/Ubuntu/Test/CMakeLists.txt
@@ -13,7 +13,7 @@ include_directories(
 
 add_library(UbuntuTestQml MODULE ${UbuntuTestQML_SOURCES})
 
-qt5_use_modules(UbuntuTestQml Core Quick Test)
+target_link_libraries(UbuntuTestQml Qt5::Core Qt5::Quick Qt5::Test)
 
 # copy qml files into build directory for shadow builds
 file(GLOB QML_JS_FILES *.js *.qml qmldir)
