diff --git a/tests/libhistoryservice/CMakeLists.txt b/tests/libhistoryservice/CMakeLists.txt
index 2b97ddc..b79ec0e 100644
--- a/tests/libhistoryservice/CMakeLists.txt
+++ b/tests/libhistoryservice/CMakeLists.txt
@@ -39,4 +39,4 @@ generate_test(EventViewTest
 # Telepathy-based tests
 generate_telepathy_test(ContactMatcherTest
                         SOURCES ContactMatcherTest.cpp
-                        QT5_MODULES Core DBus Test Qml Contacts)
+                        QT5_MODULES Qt5::Core Qt5::DBus Qt5::Test Qt5::Qml Qt5::Contacts)
diff --git a/tests/plugins/sqlite/CMakeLists.txt b/tests/plugins/sqlite/CMakeLists.txt
index f2bc520..c06f942 100644
--- a/tests/plugins/sqlite/CMakeLists.txt
+++ b/tests/plugins/sqlite/CMakeLists.txt
@@ -4,6 +4,6 @@ include_directories(
     ${CMAKE_CURRENT_BINARY_DIR}
     )
 
-generate_test(SqlitePluginTest SOURCES SqlitePluginTest.cpp LIBRARIES historyservice sqlitehistoryplugin QT5_MODULES Core DBus Test Sql)
-generate_test(SqliteThreadViewTest SOURCES SqliteThreadViewTest.cpp LIBRARIES historyservice sqlitehistoryplugin QT5_MODULES Core DBus Test Sql)
-generate_test(SqliteEventViewTest SOURCES SqliteEventViewTest.cpp LIBRARIES historyservice sqlitehistoryplugin QT5_MODULES Core DBus Test Sql)
+generate_test(SqlitePluginTest SOURCES SqlitePluginTest.cpp LIBRARIES historyservice sqlitehistoryplugin QT5_MODULES Qt5::Core Qt5::DBus Qt5::Test Qt5::Sql)
+generate_test(SqliteThreadViewTest SOURCES SqliteThreadViewTest.cpp LIBRARIES historyservice sqlitehistoryplugin QT5_MODULES Qt5::Core Qt5::DBus Qt5::Test Qt5::Sql)
+generate_test(SqliteEventViewTest SOURCES SqliteEventViewTest.cpp LIBRARIES historyservice sqlitehistoryplugin QT5_MODULES Qt5::Core Qt5::DBus Qt5::Test Qt5::Sql)
