diff --git a/src/libqtdbustest/CMakeLists.txt b/src/libqtdbustest/CMakeLists.txt
index fc162aa..a06a47e 100644
--- a/src/libqtdbustest/CMakeLists.txt
+++ b/src/libqtdbustest/CMakeLists.txt
@@ -22,11 +22,11 @@ add_library(
 	${QTDBUSTEST_SRC}
 )
 
-qt5_use_modules(
+target_link_libraries(
 	qtdbustest
-	Core
-	DBus
-	Test
+	Qt5::Core
+	Qt5::DBus
+	Qt5::Test
 )
 
 set_target_properties(
diff --git a/src/qdbus-test-runner/CMakeLists.txt b/src/qdbus-test-runner/CMakeLists.txt
index 19025ac..ce18036 100644
--- a/src/qdbus-test-runner/CMakeLists.txt
+++ b/src/qdbus-test-runner/CMakeLists.txt
@@ -9,9 +9,9 @@ target_link_libraries(
 	qtdbustest
 )
 
-qt5_use_modules(
+target_link_libraries(
 	qdbus-simple-test-runner
-	Core
+	Qt5::Core
 )
 
 install(
diff --git a/src/watchdog/CMakeLists.txt b/src/watchdog/CMakeLists.txt
index cba78e4..772397f 100644
--- a/src/watchdog/CMakeLists.txt
+++ b/src/watchdog/CMakeLists.txt
@@ -5,9 +5,9 @@ add_executable(
 	main.cpp
 )
 
-qt5_use_modules(
+target_link_libraries(
 	watchdog
-	Core
+	Qt5::Core
 )
 
 install(
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 4b0c483..c99a3e2 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -24,11 +24,11 @@ add_executable(
 	${UNIT_TESTS_SRC}
 )
 
-qt5_use_modules(
+target_link_libraries(
 	unit-tests
-	Core
-	DBus
-	Test
+	Qt5::Core
+	Qt5::DBus
+	Qt5::Test
 )
 
 target_link_libraries(
