diff --git a/data/CMakeLists.txt b/data/CMakeLists.txt
index ff16abc80..7435e16bb 100644
--- a/data/CMakeLists.txt
+++ b/data/CMakeLists.txt
@@ -28,8 +28,8 @@ install(FILES
     DESTINATION ${SHELL_APP_DIR}/Dash/
     )
 
-add_custom_target(pkgversion ALL COMMAND dpkg-parsechangelog -l${CMAKE_SOURCE_DIR}/debian/changelog --show-field version > ${CMAKE_CURRENT_BINARY_DIR}/version)
+#add_custom_target(pkgversion ALL COMMAND dpkg-parsechangelog -l${CMAKE_SOURCE_DIR}/debian/changelog --show-field version > ${CMAKE_CURRENT_BINARY_DIR}/version)
 
 install(FILES com.canonical.Unity8.gschema.xml DESTINATION ${CMAKE_INSTALL_DATADIR}/glib-2.0/schemas)
 install(FILES unity8-dash.url-dispatcher DESTINATION ${CMAKE_INSTALL_DATADIR}/url-dispatcher/urls)
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/version DESTINATION ${CMAKE_INSTALL_LOCALSTATEDIR}/lib/unity8)
+#install(FILES ${CMAKE_CURRENT_BINARY_DIR}/version DESTINATION ${CMAKE_INSTALL_LOCALSTATEDIR}/lib/unity8)
diff --git a/plugins/Lights/CMakeLists.txt b/plugins/Lights/CMakeLists.txt
index b0dedf3dc..d86f132d1 100644
--- a/plugins/Lights/CMakeLists.txt
+++ b/plugins/Lights/CMakeLists.txt
@@ -1,17 +1 @@
-pkg_check_modules(ANDROID_HEADERS REQUIRED android-headers)
-pkg_check_modules(ANDROID_HARDWARE REQUIRED libhardware)
-
-include_directories(
-    SYSTEM
-    ${ANDROID_HEADERS_INCLUDE_DIRS}
-)
-
-add_library(Lights-qml MODULE
-    plugin.cpp
-    Lights.cpp
-    )
-
-qt5_use_modules(Lights-qml Qml Gui)
-target_link_libraries(Lights-qml ${ANDROID_HARDWARE_LIBRARIES})
-
-add_unity8_plugin(Lights 0.1 Lights TARGETS Lights-qml)
+message("Sorry")
diff --git a/plugins/Ubuntu/DownloadDaemonListener/CMakeLists.txt b/plugins/Ubuntu/DownloadDaemonListener/CMakeLists.txt
index f782b821c..d86f132d1 100644
--- a/plugins/Ubuntu/DownloadDaemonListener/CMakeLists.txt
+++ b/plugins/Ubuntu/DownloadDaemonListener/CMakeLists.txt
@@ -1,18 +1 @@
-include_directories(
-    ${CMAKE_CURRENT_SOURCE_DIR}
-    ${CMAKE_CURRENT_BINARY_DIR}
-)
-
-set(DOWNLOADDAEMONLISTENER_SOURCES
-    plugin.cpp
-    DownloadTracker.cpp
-    interface/downloadtrackeradaptor.cpp
-    )
-
-add_library(DownloadDaemonListener MODULE ${DOWNLOADDAEMONLISTENER_SOURCES})
-
-qt5_use_modules(DownloadDaemonListener Qml Quick DBus Core)
-pkg_check_modules(DDLADAPTER REQUIRED udm-common)
-add_definitions(${DDLADAPTER_CFLAGS} ${DDLADAPTER_CFLAGS_OTHER})
-target_link_libraries(DownloadDaemonListener ${DDLADAPTER_LIBRARIES})
-add_unity8_plugin(Ubuntu.DownloadDaemonListener 0.1 Ubuntu/DownloadDaemonListener TARGETS DownloadDaemonListener)
+message("Sorry")
