diff --git a/test/qmltest/mocks/plugins/Unity/Application/CMakeLists.txt b/test/qmltest/mocks/plugins/Unity/Application/CMakeLists.txt
index 8dc4c6e..fd53227 100644
--- a/test/qmltest/mocks/plugins/Unity/Application/CMakeLists.txt
+++ b/test/qmltest/mocks/plugins/Unity/Application/CMakeLists.txt
@@ -20,7 +20,7 @@ set(ApplicationMocks_SOURCES
 
 add_library(ApplicationMocks SHARED ${ApplicationMocks_SOURCES})
 
-qt5_use_modules(ApplicationMocks Core Gui)
+target_link_libraries(ApplicationMocks Qt5::Core Qt5::Gui)
 
 set(TestApplicationPlugin_SOURCES
     TestApplicationPlugin.cpp
@@ -28,7 +28,7 @@ set(TestApplicationPlugin_SOURCES
 
 add_library(TestApplicationPlugin MODULE ${TestApplicationPlugin_SOURCES})
 
-qt5_use_modules(TestApplicationPlugin Core Gui Quick)
+target_link_libraries(TestApplicationPlugin Qt5::Core Qt5::Gui Qt5::Quick)
 
 target_link_libraries(TestApplicationPlugin ApplicationMocks)
 
diff --git a/test/qmltest/mocks/plugins/Unity/Launcher/CMakeLists.txt b/test/qmltest/mocks/plugins/Unity/Launcher/CMakeLists.txt
index 6d80b81..8694b48 100644
--- a/test/qmltest/mocks/plugins/Unity/Launcher/CMakeLists.txt
+++ b/test/qmltest/mocks/plugins/Unity/Launcher/CMakeLists.txt
@@ -29,7 +29,8 @@ set(LauncherMocks_SOURCES
 
 add_library(LauncherMocks SHARED ${LauncherMocks_SOURCES})
 
-qt5_use_modules(LauncherMocks Core Gui)
+#find_package(Qt5Gui REQUIRED)
+target_link_libraries(LauncherMocks Qt5::Core Qt5::Gui)
 
 set(TestLauncherPlugin_SOURCES
     TestLauncherPlugin.cpp
@@ -37,7 +38,7 @@ set(TestLauncherPlugin_SOURCES
 
 add_library(TestLauncherPlugin MODULE ${TestLauncherPlugin_SOURCES})
 
-qt5_use_modules(TestLauncherPlugin Core Quick)
+target_link_libraries(TestLauncherPlugin Qt5::Core Qt5::Quick)
 
 target_link_libraries(TestLauncherPlugin LauncherMocks)
 
diff --git a/test/qmltest/mocks/plugins/Unity/Notifications/CMakeLists.txt b/test/qmltest/mocks/plugins/Unity/Notifications/CMakeLists.txt
index e7b6bd5..2054736 100644
--- a/test/qmltest/mocks/plugins/Unity/Notifications/CMakeLists.txt
+++ b/test/qmltest/mocks/plugins/Unity/Notifications/CMakeLists.txt
@@ -21,7 +21,7 @@ set(NotificationsMocks_SOURCES
 
 add_library(NotificationsMocks SHARED ${NotificationsMocks_SOURCES})
 
-qt5_use_modules(NotificationsMocks Core)
+target_link_libraries(NotificationsMocks Qt5::Core)
 
 set(TestNotificationsPlugin_SOURCES
     TestNotificationsPlugin.cpp
@@ -29,7 +29,7 @@ set(TestNotificationsPlugin_SOURCES
 
 add_library(TestNotificationsPlugin MODULE ${TestNotificationsPlugin_SOURCES})
 
-qt5_use_modules(TestNotificationsPlugin Core Quick)
+target_link_libraries(TestNotificationsPlugin Qt5::Core Qt5::Quick)
 
 target_link_libraries(TestNotificationsPlugin NotificationsMocks)
 
diff --git a/test/qmltest/mocks/plugins/Unity/Notifications/Mocks/CMakeLists.txt b/test/qmltest/mocks/plugins/Unity/Notifications/Mocks/CMakeLists.txt
index 221ddaf..d8b85c0 100644
--- a/test/qmltest/mocks/plugins/Unity/Notifications/Mocks/CMakeLists.txt
+++ b/test/qmltest/mocks/plugins/Unity/Notifications/Mocks/CMakeLists.txt
@@ -4,7 +4,9 @@ set(MockNotificationsPlugin_SOURCES
 
 add_library(MockNotificationsPlugin MODULE ${MockNotificationsPlugin_SOURCES})
 
-qt5_use_modules(MockNotificationsPlugin Core Quick)
+#find_package(Qt5Core REQUIRED)
+#find_package(Qt5Quick REQUIRED)
+target_link_libraries(MockNotificationsPlugin Qt5::Core Qt5::Quick)
 
 target_link_libraries(MockNotificationsPlugin NotificationsMocks)
 
diff --git a/test/qmltest/mocks/plugins/Unity/Scopes/CMakeLists.txt b/test/qmltest/mocks/plugins/Unity/Scopes/CMakeLists.txt
index 452aec2..c06a361 100644
--- a/test/qmltest/mocks/plugins/Unity/Scopes/CMakeLists.txt
+++ b/test/qmltest/mocks/plugins/Unity/Scopes/CMakeLists.txt
@@ -39,7 +39,7 @@ set(ScopesMocks_SOURCES
 
 add_library(ScopesMocks SHARED ${ScopesMocks_SOURCES})
 
-qt5_use_modules(ScopesMocks Core)
+target_link_libraries(ScopesMocks Qt5::Core)
 
 set(TestScopesPlugin_SOURCES
     TestScopesPlugin.cpp
@@ -47,7 +47,7 @@ set(TestScopesPlugin_SOURCES
 
 add_library(TestScopesPlugin MODULE ${TestScopesPlugin_SOURCES})
 
-qt5_use_modules(TestScopesPlugin Core Quick)
+target_link_libraries(TestScopesPlugin Qt5::Core Qt5::Quick)
 
 target_link_libraries(TestScopesPlugin ScopesMocks)
 
diff --git a/test/qmltest/modules/TestUtil/CMakeLists.txt b/test/qmltest/modules/TestUtil/CMakeLists.txt
index d4fb358..8a812f5 100644
--- a/test/qmltest/modules/TestUtil/CMakeLists.txt
+++ b/test/qmltest/modules/TestUtil/CMakeLists.txt
@@ -15,7 +15,7 @@ include_directories(
 
 add_library(TestUtilQml MODULE ${TestUtilQML_SOURCES})
 
-qt5_use_modules(TestUtilQml Core Quick)
+target_link_libraries(TestUtilQml Qt5::Core Qt5::Quick)
 
 # copy qmldir file into build directory for shadow builds
 file(GLOB QML_JS_FILES qmldir *.js *.qml)
