diff --git a/CMakeLists.txt b/CMakeLists.txt
index 22479d9..0e86292 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,7 +18,7 @@ if (NOT DEFINED TRUST_STORE_VERSION_MAJOR)
   #   - other distros
   #   - errors
   # we define the version to be 2.0.0
-  if (${DISTRO_CODENAME} STREQUAL "vivid")
+  if (DISTRO_CODENAME STREQUAL "vivid")
     set(TRUST_STORE_VERSION_MAJOR 1)
     set(TRUST_STORE_VERSION_MINOR 1)
     set(TRUST_STORE_VERSION_PATCH 0) 
