diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index f58fbd1..dcc2927 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -9,14 +9,14 @@ find_package(GMock)
 add_executable (helper-test helper-test.cc)
 add_definitions ( -DCMAKE_SOURCE_DIR="${CMAKE_CURRENT_SOURCE_DIR}" )
 add_definitions ( -DCMAKE_BINARY_DIR="${CMAKE_CURRENT_BINARY_DIR}" )
-target_link_libraries (helper-test launcher-static gtest_main ${GTEST_MAIN_LIBRARIES} ${DBUSTEST_LIBRARIES})
+target_link_libraries (helper-test launcher-static gtest_main ${GMOCK_LIBRARIES} ${DBUSTEST_LIBRARIES})
 
 add_test (helper-test helper-test)
 
 # Helper test
 
 add_executable (helper-handshake-test helper-handshake-test.cc)
-target_link_libraries (helper-handshake-test launcher-static gtest_main ${GTEST_MAIN_LIBRARIES})
+target_link_libraries (helper-handshake-test launcher-static gtest_main ${GMOCK_LIBRARIES})
 
 add_test (helper-handshake-test helper-handshake-test)
 
@@ -33,7 +33,7 @@ add_definitions ( -DSNAP_BASEDIR="${CMAKE_CURRENT_SOURCE_DIR}/snap-basedir" )
 add_executable (libual-test
 	libual-test.cc
 	mir-mock.cpp)
-target_link_libraries (libual-test gtest_main ${GTEST_MAIN_LIBRARIES} ${LIBUPSTART_LIBRARIES} ${DBUSTEST_LIBRARIES} launcher-static)
+target_link_libraries (libual-test gtest_main ${GMOCK_LIBRARIES} ${LIBUPSTART_LIBRARIES} ${DBUSTEST_LIBRARIES} launcher-static)
 
 add_executable (libual-cpp-test
 	libual-cpp-test.cc
@@ -54,7 +54,7 @@ add_test (NAME libual-cpp-test COMMAND libual-cpp-test)
 
 add_executable (app-store-legacy
 	app-store-legacy.cpp)
-target_link_libraries (app-store-legacy ${GMOCK_LIBRARIES} ${GTEST_MAIN_LIBRARIES} launcher-static ${DBUSTEST_LIBRARIES})
+target_link_libraries (app-store-legacy ${GMOCK_LIBRARIES} ${GMOCK_LIBRARIES} launcher-static ${DBUSTEST_LIBRARIES})
 
 add_test(NAME app-store-legacy COMMAND "${CMAKE_CURRENT_BINARY_DIR}/app-store-legacy" --gtest_list_tests "|" grep "\"^  \"" "|" xargs -n 1 printf "\"--gtest_filter=*.%s\\n\"" "|" xargs -n 1 "${CMAKE_CURRENT_BINARY_DIR}/app-store-legacy")
 
@@ -63,7 +63,7 @@ add_test(NAME app-store-legacy COMMAND "${CMAKE_CURRENT_BINARY_DIR}/app-store-le
 
 add_executable (jobs-base-test
 	jobs-base-test.cpp)
-target_link_libraries (jobs-base-test ${GMOCK_LIBRARIES} ${GTEST_MAIN_LIBRARIES} launcher-static ${DBUSTEST_LIBRARIES})
+target_link_libraries (jobs-base-test ${GMOCK_LIBRARIES} ${GMOCK_LIBRARIES} launcher-static ${DBUSTEST_LIBRARIES})
 
 add_test(NAME jobs-base-test COMMAND jobs-base-test)
 
@@ -89,7 +89,7 @@ if(CURL_FOUND)
 add_definitions ( -DSNAPD_TEST_SOCKET="/tmp/snapd-test-socket" )
 add_executable (snapd-info-test
 	snapd-info-test.cpp)
-target_link_libraries (snapd-info-test gtest_main ${GTEST_MAIN_LIBRARIES} launcher-static)
+target_link_libraries (snapd-info-test gtest_main ${GMOCK_LIBRARIES} launcher-static)
 add_test (NAME snapd-info-test COMMAND snapd-info-test)
 endif()
 
@@ -97,7 +97,7 @@ endif()
 
 add_executable (list-apps
 	list-apps.cpp)
-target_link_libraries (list-apps gtest_main ${GTEST_MAIN_LIBRARIES} ${DBUSTEST_LIBRARIES} launcher-static)
+target_link_libraries (list-apps gtest_main ${GMOCK_LIBRARIES} ${DBUSTEST_LIBRARIES} launcher-static)
 add_test (NAME list-apps COMMAND ${CMAKE_CURRENT_BINARY_DIR}/list-apps)
 
 # Application Info Desktop
@@ -117,7 +117,7 @@ add_executable (application-icon-finder-test
 
   #sources
   ${CMAKE_SOURCE_DIR}/libubuntu-app-launch/application-icon-finder.cpp)
-target_link_libraries (application-icon-finder-test gtest_main ${GTEST_MAIN_LIBRARIES} ubuntu-launcher)
+target_link_libraries (application-icon-finder-test gtest_main ${GMOCK_LIBRARIES} ubuntu-launcher)
 
 add_test (NAME application-icon-finder-test COMMAND application-icon-finder-test)
 
