diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8ae44ba..7492536 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -88,12 +88,6 @@ include_directories(${LTTNG_INCLUDE_DIRS})
 pkg_check_modules(MIR REQUIRED mirclient)
 include_directories(${MIR_INCLUDE_DIRS})
 
-pkg_check_modules(LIBERTINE libertine)
-include_directories(${LIBERTINE_INCLUDE_DIRS})
-if(${LIBERTINE_FOUND})
-  add_definitions(-DHAVE_LIBERTINE=1)
-endif(${LIBERTINE_FOUND})
-
 pkg_check_modules(CURL libcurl>=7.47)
 include_directories(${CURL_INCLUDE_DIRS})
 
